src/check_bgp_protocols.py

```diff
@@ -69,27 +69,24 @@ def check_birdc_peers(path):
             ExitCodes.critical, 'Could not retrieve BIRD\'s output'
         )
 
-    try:
-        r = re.compile(
-            r'^(?P<name>\w+)\s+BGP\s+(\w+)\s+(?P<state>\w+)\s+'
-            r'(?P<since>[\d\-:]+)\s+(?P<info>[\w]+)?\s*$',
-            re.MULTILINE
-        )
+    r = re.compile(
+        r'^(?P<name>\w+)\s+BGP\s+(\w+)\s+(?P<state>\w+)\s+'
+        r'(?P<since>[\d\-:]+)\s+(?P<info>[\w]+)?\s*$',
+        re.MULTILINE
+    )
 
-        res = [m.groupdict() for m in r.finditer(protocols)]
-    except Exception:
-        return ExitCodes.critical, 'Could not parse BIRD\'s output'
+    res = [m.groupdict() for m in r.finditer(protocols)]
```

Comment:
Now I see it! It should break now. You were safe before because of the `except`. `m.groupdict()` was throwing an Exception for None items whenever `finditer` does not match.
Please check again my theory with your script. If it is breaking now, you can easily change it with:
```suggestion
    res = [
        m.groupdict() for m in r.finditer(protocols)
        if m is not None
    ]
```

Reply:
I have tested this extensively with some crafted output and couldn't hit any case where it contained None. 
I tried with empty lines, lines with just a space and couldn't make it happen. I think the `if m is not None` is a small change and doesn't really hurt, so I'll add it just in case.

Reply:
Doesn't apply anymore